### PR TITLE
Migrate to TailwindCSS v3

### PR DESCRIPTION
**Pull request by robdekort**
Upgrade to TailwindCSS v3

- [x] Upgrade packages 
- [x] Update content configuration
- [x] Remove redundant utilities
- [x] Update colors
- [x] Use `file:` modifier
- [x] Rename utilities
- [x] Make Aspect Ratio plugin work
- [x] Make SVG utilities compile

Running into: https://github.com/tailwindlabs/tailwindcss/discussions/6310

### CODE REVIEW

**Comment by robdekort** [comment]
Ok, I've updated the configuration order to fix those issues BUT now all TW colors are available in JIT which I don't really want.

**Comment by sjclark** [comment]
> Ok, I've updated the configuration order to fix those issues BUT now all TW colors are available in JIT which I don't really want.

Can understand - but I guess thats really the **entire** approach of Tailwind now - you get EVEYRTHING whether you want it or not. Particularly now that they've removed variants etc. Just because they're there doesn't mean you have to use them (I'm sure you don't use 99% of the other classes already?) 🙂

**Comment by robdekort** [comment]
I don't agree. This is why you can customize tailwind. It doesn't make sense to enable everything when you're working within the constraints of design system. 

In this case it seems like a misconfiguration or bug.

**Comment by jelleroorda** [comment]
I think like @sjclark already said that’s how the new tailwind vision is. So it’s not a bug, but I can imagine that you want to enforce a specific design system. In that case you can either document what should be used, or you need a code review (or design document) I suppose.

File: dev/tailwind.config.site.js [hunk 1]
```diff
@@ -12,23 +12,23 @@ const colors = require('tailwindcss/colors')
 
 module.exports = {
   theme: {
-    // Here we define the default colors available. If you want to include
-    // all default Tailwind colors you should extend the colors instead.
-    colors: {
```
Comment (jelleroorda) [inline dev/tailwind.config.site.js:17]: I noticed you moved this out of the colors and into extend though, was that on purpose? Extend should extend and not constrain.

**Comment by robdekort** [comment]
https://tailwindcss.com/docs/customizing-colors#using-custom-colors

You can still replace the default color set. It's what I've always done in Peak and it's still be possible. Including all colors is just the default in TW v3. 

However. I did change the colors in the Peak config to extend this morning. I'm basically testing what works and what not. At least currently everything works but it includes all colors :-)

**Comment by robdekort** [comment]
Ah, wait a minute, you have to define 'currentColor' now. That's probably why - when overwriting the default color config - the fill utilities weren't working before.

**Comment by robdekort** [comment]
Ok, I think I solved this now.

**Comment by robdekort** [comment]
They only thing I _had_ to change though was move:

```js
require('@tailwindcss/aspect-ratio'),
require('@tailwindcss/forms'),
```

From `tailwind.config.peak.js` to `tailwind.config.js`. I don't like as it belongs in the original file, but it seems the aspect ratio plugin doesn't work when it gets required in the other (last) file.

**Comment by robdekort** [comment]
> They only thing I _had_ to change though was move:
> 
> ```js
> require('@tailwindcss/aspect-ratio'),
> require('@tailwindcss/forms'),
> ```
> 
> From `tailwind.config.peak.js` to `tailwind.config.js`. I don't like as it belongs in the original file, but it seems the aspect ratio plugin doesn't work when it gets required in the other (last) file.

This is also back where it belongs thanks to a nice tip and explanation from Adam.

**Comment by sjclark** [comment]
> > They only thing I _had_ to change though was move:
> > ```js
> > require('@tailwindcss/aspect-ratio'),
> > require('@tailwindcss/forms'),
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > From `tailwind.config.peak.js` to `tailwind.config.js`. I don't like as it belongs in the original file, but it seems the aspect ratio plugin doesn't work when it gets required in the other (last) file.
> 
> This is also back where it belongs thanks to a nice tip and explanation from Adam.

Can you link to the tip / explanation? (I still struggling with some weird edge issues on things so keen to find any documentation of anything!)

**Comment by robdekort** [comment]
https://github.com/tailwindlabs/tailwindcss-aspect-ratio/issues/30

Hope it helps.

**Pull request merged**